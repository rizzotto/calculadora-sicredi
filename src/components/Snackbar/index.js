import React from "react";
import { useDispatch, useSelector } from "react-redux";

import { Snackbar as MuiSnackbar } from "@material-ui/core";
import MuiAlert from "@material-ui/lab/Alert";
import PropTypes from "prop-types";

import { closeSnackbar } from "../../actions/snackbarActions.js";

function Alert(props) {
    return <MuiAlert elevation={6} variant="filled" {...props} />;
}

export default function Snackbar() {
    const dispatch = useDispatch();
    const state = useSelector((state) => state.snackbar);

    const { open, message, severity } = state;

    const handleClose = (event, reason) => {
        if (reason === "clickaway") {
            return;
        }
        dispatch(closeSnackbar());
    };

    return (
        <MuiSnackbar
            open={open}
            onClose={handleClose}
            anchorOrigin={{ vertical: "bottom", horizontal: "right" }}
            autoHideDuration={5000}
        >
            <Alert onClose={handleClose} severity={severity}>
                {message}
            </Alert>
        </MuiSnackbar>
    );
}

Snackbar.propTypes = {
    severity: PropTypes.string,
    open: PropTypes.bool,
    errorMessage: PropTypes.string,
    onClose: PropTypes.func,
};
