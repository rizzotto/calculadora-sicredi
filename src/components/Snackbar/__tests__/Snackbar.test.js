import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";

import store from "../../../providers/Store.js";
import Snackbar from "../index.js";

it("renders Snackbar correctly", () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <Snackbar />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
