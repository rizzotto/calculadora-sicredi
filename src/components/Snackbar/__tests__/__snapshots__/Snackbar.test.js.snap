// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Snackbar correctly 1`] = `null`;
